flybywiresim/a32nx
# fix: Improve import flight plan from MSFS World Map

## Summary of Changes
Currently the import from the MSFS World Map is officially supported when the The SimOptions setting "Sync MSFS Flight Plan" is set to `Load Only` or `Save`.

Unfortunately this import is still unreliable and needs imrpovement for users who prefer the MSFS World Map over SimBrief.

Current know issues:

- [x] Order of imported waypoint is wrong
- [x] Waypoints are missing (not imported)
- [x] fix eslint error as far as useful

## Additional context
https://github.com/flybywiresim/docs/pull/691

Test 1:
![image](https://user-images.githubusercontent.com/16833201/206843820-2d75ae9c-6460-4fbd-856d-19889dd063b5.png)
![image](https://user-images.githubusercontent.com/16833201/206843840-407ab25b-66d0-4f7e-b12a-813d509b36ac.png)
`LPPT ROKOB BOMKO ALAMA FTM NEA VRA SSN URUNA IBDER  REPSI  MEBAK  NINTU  MILPA  VADAR  FRI  SUREP  DITON  KUDIS  NUNRI  BETOS  EDDM`

Test 2:
EIDW DE28L D1 DW080 DW682 NEPOD ABIDO PESIT GWC WAFFU ELDAX PIREG SOMIL IRBAL RANUX NEBAX OBAKI BEGAR ELMUR RESIA UNTAD DIKEM ROKIB BABAG PEVAL PUL INTIT MD326 NEVRA KRO LGAV

Discord username (if different from GitHub): Cdr_Maverick#6475

## Testing instructions
- Plan a flight via the MSFS World Map
- Use various combinations of "only enroute", with/without SID, with/without STAR/APPR, etc.
- Test from Gate and Runway (mid air flight plan loading is not yet supported)

OBS: it might be required to enter SID/STAR/APPR in the MCDU again - esp. the STAR/APPR will sometimes not be loaded as expected as the MSFS flight plan does not really handle all arrival and approach combinations (e.g. VIAs).

Tips:
- make screenshots or notes of the planned route from World Map
- compare it to the MCDU F-PLN
- iterate through the MCDU F_PLN on the ND

Updated Docs:
https://github.com/flybywiresim/docs/pull/691


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
